### ISSUE TITLE
Unsupported playlists when run from WAR / container, not when running `org.airsonic.player.Application` exploded

### ISSUE THREAD

**Issue by Rudd-O** [airsonic-advanced/airsonic-advanced#331]
Using latest EDGE:

```
Jun 23 13:11:03 penny.dragonfear java[893543]: 2020-06-23 13:11:03.938  WARN --- o.a.p.service.PlaylistService            : Failed to auto-import playlist /srv/shared/Entertainment/Music/Airsonic playlists/2012-03.m3u
Jun 23 13:11:03 penny.dragonfear java[893543]: java.lang.Exception: Unsupported playlist 2012-03.m3u
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at org.airsonic.player.service.PlaylistService.importPlaylist(PlaylistService.java:239) ~[classes!/:11.0.0-SNAPSHOT]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at org.airsonic.player.service.PlaylistService.importPlaylistIfUpdated(PlaylistService.java:352) ~[classes!/:11.0.0-SNAPSHOT]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at org.airsonic.player.service.PlaylistService.lambda$15(PlaylistService.java:329) ~[classes!/:11.0.0-SNAPSHOT]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at org.airsonic.player.service.PlaylistService.doImportPlaylists(PlaylistService.java:327) ~[classes!/:11.0.0-SNAPSHOT]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at org.airsonic.player.service.PlaylistService.importPlaylists(PlaylistService.java:308) ~[classes!/:11.0.0-SNAPSHOT]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at org.airsonic.player.service.MediaScannerService.lambda$3(MediaScannerService.java:195) ~[classes!/:11.0.0-SNAPSHOT]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(CompletableFuture.java:783) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594) ~[na:na]
Jun 23 13:11:03 penny.dragonfear java[893543]: 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177) ~[na:na]
```

Airsonic normally supports these.

**Comment by Rudd-O** [comment]
```
        final SpecificPlaylist inputSpecificPlaylist = SpecificPlaylistFactory.getInstance().readFrom(inputStream, "UTF-8");
        if (inputSpecificPlaylist == null) {
            throw new Exception("Unsupported playlist " + fileName);
        }
```

My playlists are all UTF-8.

**Comment by randomnicode** [comment]
Hmmm, don't think this code has logically changed from vanilla airsonic.

perhaps you could post the m3u here?

**Comment by sandersantema** [comment]
For me it worked after restarting the server and changing the extension from m3u to m3u8 but I don't know if this had anything to do with the playlist import working/ automatic update of playlists.

**Comment by MickMorley** [comment]
I experienced this when I first loaded the docker version.  I manually uploaded the same file it was trying to import and it worked.  After that for some reason the automatic import worked.

I've never had any playlist import issues with the vanilla airsonic.

**Comment by SyntrophicConsortium** [comment]
I'm seeing the same error with an .m3u. I'm in the process of doing some troubleshooting to determine if the filesystem is the issue (music is on ntfs, but the playlist file is on ext4). I'll try renaming to .m3u8 to see if that makes a difference, though I'm not sure why that would make a difference.

**Comment by lbesnard** [comment]
I do have the same issue. I updated the path of the files inside the playlist, make sure the path are correct. both auto and manual import of the playlists fail

**Comment by Rudd-O** [comment]
The issue persists with latest EDGE release.

**Comment by Rudd-O** [comment]
I've tried m3u8 and xspf, both valid playlists.  They don't work either.

**Comment by Rudd-O** [comment]
I've also tried playlists with exactly one, valid, found, absolute path.  They do not work either.

**Comment by Rudd-O** [comment]
Bear in mind this happens with latest EDGE as well.

**Comment by Rudd-O** [comment]
Docker built version does not work either.

```
[pid 1384329] access("/srv/shared/Entertainment/Music/Playlists/2020-12.m3u", R_OK) = 0
[pid 1384329] openat(AT_FDCWD, "/srv/shared/Entertainment/Music/Playlists/2020-12.m3u", O_RDONLY) = 24
```

Files ARE being accessed normally by Airsonic.

**Comment by rigrig** [comment]
In case it helps:

That `java.lang.Exception: Unsupported playlist Best of Live Radiohead.m3u` is thrown at https://github.com/airsonic-advanced/airsonic-advanced/blob/528e587fe8198c6c7e74d927a5820a1cceb81dfc/airsonic-main/src/main/java/org/airsonic/player/service/PlaylistService.java#L237-L239

`SpecificPlaylistFactory` is provided by https://github.com/muff1nman/chameleon
https://github.com/airsonic-advanced/airsonic-advanced/blob/528e587fe8198c6c7e74d927a5820a1cceb81dfc/airsonic-main/src/main/java/org/airsonic/player/service/PlaylistService.java#L23

And tries to load the playlist here: https://github.com/muff1nman/chameleon/blob/bc95a3f1dcfd3cd076a593cdb569069d5440cc01/core/src/main/java/chameleon/playlist/SpecificPlaylistFactory.java#L130

It looks like https://github.com/muff1nman/chameleon/blob/bc95a3f1dcfd3cd076a593cdb569069d5440cc01/core/src/main/java/chameleon/playlist/SpecificPlaylistFactory.java#L139-L142 returns null:
```java
                InputStream temp = new ByteArrayInputStream(bytes);
                ret = service.readFrom(temp, contentEncoding, _logger); // May throw Exception. Shall not throw NullPointerException because of in.
                // Returns it even if null.
                break;
```
So apparently `M3uPlaylist::readFrom` throws an error somewhere in https://github.com/muff1nman/chameleon/blob/bc95a3f1dcfd3cd076a593cdb569069d5440cc01/playlists/m3u/src/main/java/chameleon/playlist/m3u/M3UProvider.java#L109


Which is weird, because all this code seems to be untouched for the last couple of years, so you would expect it to just keep working...

**Comment by Rudd-O** [comment]
Yes, extremely weird.  I went through the code too and that's exactly the code path that it takes.

I also noticed that Chameleon has not changed in three years, even their new 1.3 snapshot is only version number bump.

Something is iffy -- perhaps there's something in how the Input Stream is fed to SpecificPlaylistFactory?  I couldn't get Chameleon to emit debugging anything.  What I suspect is something about the packaging is making Chameleon not load its providers, that could be the only explanation why none of the providers emit logs.

**Comment by Rudd-O** [comment]
**HOLY SHIT!**

I deployed Airsonic advanced, NOT from the WAR, but exploded, and PLAYLIST IMPORT WORKS!!!  I followed this process:

https://github.com/airsonic-advanced/airsonic-advanced/pull/315/files

Please look into it!

There's something wrong with the packaging of the WAR.  I can't tell what it is.

**Comment by Rudd-O** [comment]
Separate bug: My Audinaut app shows me all playlists and loads them correctly.  The Web UI simply shows _Processing_ where the songs should load.  The web console says `Uncaught ReferenceError: formatDuration is not defined`

```
render http://penny:8080/playlist.view?id=89:220
    U http://penny:8080/script/DataTables/datatables.min.js:43
    fnGetData http://penny:8080/script/DataTables/datatables.min.js:37
    I http://penny:8080/script/DataTables/datatables.min.js:42
    Lb http://penny:8080/script/DataTables/datatables.min.js:90
    Kb http://penny:8080/script/DataTables/datatables.min.js:90
    Ja http://penny:8080/script/DataTables/datatables.min.js:88
    aa http://penny:8080/script/DataTables/datatables.min.js:38
    b http://penny:8080/script/DataTables/datatables.min.js:90
    e http://penny:8080/script/DataTables/datatables.min.js:121
    setTimeout handler*throttle/< http://penny:8080/script/DataTables/datatables.min.js:121
```

**Comment by Rudd-O** [comment]
Strike the above about the web stuff failing — I just needed to refresh my browser cache — there's probably a way to serve a different cache key for static files upon restart.

**Comment by randomnicode** [comment]
This issue should be fixed with #539 

May we please verify and close this issue?

**Comment by randomnicode** [comment]
Closing this issue since this seems resolved now
